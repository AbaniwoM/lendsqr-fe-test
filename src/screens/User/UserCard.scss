.userCard {
    display: flex;
    flex-direction: column;
    justify-content: center;
    background-color: white;
    gap: 5px;
    padding-left: 25px;
    width: 19.5%;
    height: 20.5vh;
}

.userImg img {
    width: 18%;
    height: 6vh;
}

.userDesc {
   color: #545F7D;
   font-size: 0.8rem;
   font-weight: 500;
}

.userCount {
   color: #213F7D;
   font-size: 1.2rem;
   font-weight: 700;
}