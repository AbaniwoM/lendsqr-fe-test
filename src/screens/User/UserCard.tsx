import React from 'react';
import "./UserCard.scss";

interface UserCardProps {
    img: string;
    desc: string;
    count: number;
}

const UserCard = ({ img, desc, count}: UserCardProps) => {
  return (
    <div className="userCard" data-testid="userCard-1">
        <div className="userImg" data-testid="userImg-1">
           <img src={img} alt={"img"} />
        </div>
        <div className="userDesc" data-testid="userDesc-1">{desc}</div>
        <div className="userCount" data-testid="userCount-1">{count}</div>
    </div>
  )
}

export default UserCard;