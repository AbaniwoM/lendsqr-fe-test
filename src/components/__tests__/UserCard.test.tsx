import { render, screen, cleanup } from '@testing-library/react';
// import Dashboard from '../../screens/Dashboard/Dashboard';
import UserCard from '../../screens/User/UserCard';
import renderer from 'react-test-renderer';

afterEach(() => {
    cleanup();
});

test('should render userCard component', () => {
    render(<UserCard img={''} desc={''} count={0} />);
    const userCardElement = screen.getByTestId('userCard-1');
    expect(userCardElement).toBeInTheDocument();
})

test('should render userImg class', () => {
    render(<UserCard img={''} desc={''} count={0} />);
    const userImgClass = screen.getByTestId('userImg-1');
    expect(userImgClass).toBeInTheDocument();
    expect(userImgClass).toHaveClass("userImg");
});

test('should render userDesc class', () => {
    render(<UserCard img={''} desc={''} count={0} />);
    const userDescClass = screen.getByTestId('userDesc-1');
    expect(userDescClass).toBeInTheDocument();
    expect(userDescClass).toHaveClass("userDesc");
});

test('should render userCount class', () => {
    render(<UserCard img={''} desc={''} count={0} />);
    const userCountClass = screen.getByTestId('userCount-1');
    expect(userCountClass).toBeInTheDocument();
    expect(userCountClass).toHaveClass("userCount");
});

test('matches snapshot', () => {
    const tree = renderer.create(<UserCard img={''} desc={''} count={0} />).toJSON();
    expect(tree).toMatchSnapshot();
})