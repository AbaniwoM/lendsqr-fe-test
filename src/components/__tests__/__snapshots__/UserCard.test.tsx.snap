// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="userCard"
  data-testid="userCard-1"
>
  <div
    className="userImg"
    data-testid="userImg-1"
  >
    <img
      alt="img"
      src=""
    />
  </div>
  <div
    className="userDesc"
    data-testid="userDesc-1"
  />
  <div
    className="userCount"
    data-testid="userCount-1"
  >
    0
  </div>
</div>
`;
